packages/edit-navigation/src/components/menus-editor/index.js

```diff
@@ -19,33 +25,64 @@ import CreateMenuArea from './create-menu-area';
 import NavigationEditor from '../navigation-editor';
 
 export default function MenusEditor( { blockEditorSettings } ) {
-	const { menus, hasLoadedMenus } = useSelect( ( select ) => {
-		const { getMenus, hasFinishedResolution } = select( 'core' );
-		const query = { per_page: -1 };
-		return {
-			menus: getMenus( query ),
-			hasLoadedMenus: hasFinishedResolution( 'getMenus', [ query ] ),
-		};
-	}, [] );
-
+	const [ menuId, setMenuId ] = useState();
+	const [ showCreateMenuPanel, setShowCreateMenuPanel ] = useState( false );
 	const [ hasCompletedFirstLoad, setHasCompletedFirstLoad ] = useState(
 		false
 	);
 
+	const noticeId = useRef();
+
+	const { menus, hasLoadedMenus, menuDeleteError } = useSelect(
+		( select ) => {
+			const {
+				getMenus,
+				hasFinishedResolution,
+				getLastEntityDeleteError,
+			} = select( 'core' );
+			const query = { per_page: -1 };
+			return {
+				menus: getMenus( query ),
+				hasLoadedMenus: hasFinishedResolution( 'getMenus', [ query ] ),
+				menuDeleteError: getLastEntityDeleteError(
+					'root',
+					'menu',
+					menuId
+				),
+			};
+		},
+		[ menuId ]
+	);
+
+	const { deleteMenu } = useDispatch( 'core' );
+	const { createErrorNotice, removeNotice } = useDispatch( 'core/notices' );
+
 	useEffect( () => {
 		if ( ! hasCompletedFirstLoad && hasLoadedMenus ) {
 			setHasCompletedFirstLoad( true );
 		}
 	}, [ hasLoadedMenus ] );
 
-	const [ menuId, setMenuId ] = useState();
-	const [ stateMenus, setStateMenus ] = useState();
-	const [ showCreateMenuPanel, setShowCreateMenuPanel ] = useState( false );
+	// Handle REST API Error messages.
+	useEffect( () => {
+		if ( menuDeleteError ) {
+			// Error messages from the REST API often contain HTML.
+			// createErrorNotice does not support HTML in error text, so first
+			// strip HTML out using DOMParser.
+			const document = new DOMParser().parseFromString(
+				menuDeleteError.message,
+				'text/html'
+			);
+			const errorText = document.body.textContent || '';
+			noticeId.current = uniqueId(
+				'navigation-editor/menu-editor/edit-navigation-delete-menu-error'
+			);
+			createErrorNotice( errorText, { id: noticeId.current } );
+		}
+	}, [ menuDeleteError ] );
```

Comment:
Why recreate the notice ID when `menuDeleteError` changes? Will `removeNotice` always reference the right ID with this approach?

Reply:
If the errors are the same they should not stack but if they're different they should. I think this is accomplished this way.